alpha/lib/model/widget.php

```diff
@@ -276,4 +276,13 @@ public function getIsPlayList (  )	{		return $this->getFromCustomData( "isPlayLi
 	
 	public function setAddEmbedHtml5Support ( $v )	{	$this->putInCustomData ( "addEmbedHtml5Support" , (bool) $v );	}
 	public function getAddEmbedHtml5Support (  )	{	return (bool) $this->getFromCustomData( "addEmbedHtml5Support" ,null, false );	}
+
+	// set a widget role to allow a specific widget to have extra privileges
+	public function setRoles ( $v )	{	$this->putInCustomData ( "roles" , $v );	}
+	public function getRoles (  )	{	return $this->getFromCustomData( "roles" );	}
+
+	// limit operations done with a session generated with this Widget to a specific entry
+	public function setLimitEntry ( $v )	{	$this->putInCustomData ( "limitEntry" , $v );	}
```

Comment:
limit-entry attribute is not needed, if entry-id attribute is set you can know that you should limit-entry.